upcloud/service/kubernetes_test.go

```diff
@@ -2,304 +2,473 @@ package service
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"testing"
 	"time"
 
 	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud/client"
 	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud/request"
-	"github.com/dnaeon/go-vcr/recorder"
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestKubernetes(t *testing.T) {
```

Comment:
This diff is really hard to read. Basically what I did is entirely deleted old tests based on recorder and wrote new ones with `httptest.Server`

Reply:
:+1:  [Convenience link](https://github.com/UpCloudLtd/upcloud-go-api/blob/feat/k8s-node-group-aa/upcloud/service/kubernetes_test.go) to the file